clevertap-core/src/main/java/com/clevertap/android/sdk/CTXtensions.kt

```diff
@@ -299,3 +305,17 @@ fun String?.toJsonOrNull(): JSONObject? {
         }
     }
 }
+
+fun View.applySystemBarsInsetsWithMargin(block : (bars:Insets, mlp:MarginLayoutParams) -> Unit) {
+    ViewCompat.setOnApplyWindowInsetsListener(this
+    ) { v, insets ->
+        val bars: Insets = insets.getInsets(
+            WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout()
+        )
+        v.updateLayoutParams<MarginLayoutParams> {
+            block(bars,this)
+        }
+        WindowInsetsCompat.CONSUMED
```

Comment:
why are we consuming the insets, if out work is done we can let it flow to view heirarchy.

Reply:
To avoid performance overhead and as given in official docs https://developer.android.com/develop/ui/views/layout/edge-to-edge#cutout-insets